trin-core/src/portalnet/overlay.rs

```diff
@@ -175,12 +190,182 @@ impl<TContentKey: OverlayContentKey + Send, TMetric: Metric + Send>
     }
 
     /// Process overlay uTP listener event
-    pub fn process_utp_event(&self, event: UtpListenerEvent) {
-        // TODO: Handle overlay uTP events
-        debug!(
-            "Got overlay uTP event: {event:?}, protocol id: {:?}",
-            self.protocol
-        );
+    pub fn process_utp_event(&self, event: UtpListenerEvent) -> anyhow::Result<()> {
+        match event {
+            UtpListenerEvent::ClosedStream(utp_payload, _, stream_id) => {
+                // Handle all closed uTP streams, currently we process only AcceptStream here.
+                // FindContent payload is processed explicitly when we send FindContent request.
+                if let AcceptStream(content_keys) = stream_id {
+                    self.process_accept_utp_payload(content_keys, utp_payload)?;
+                }
+            }
+            UtpListenerEvent::ResetStream(..) => {
+                // TODO: Handle reset streams
```

Comment:
Was this `TODO` left here purposefully? If so, why can't we handle this now?

Reply:
Same answer as https://github.com/ethereum/trin/pull/328/files#r894254880